luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/SingleDataSourcePhysicalTableFactory.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.config.luthier.factories;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.yahoo.bard.webservice.config.luthier.Factory;
+import com.yahoo.bard.webservice.config.luthier.LuthierIndustrialPark;
+import com.yahoo.bard.webservice.config.luthier.LuthierValidationUtils;
+import com.yahoo.bard.webservice.data.config.LuthierPhysicalTableParams;
+import com.yahoo.bard.webservice.data.config.LuthierTableName;
+import com.yahoo.bard.webservice.data.dimension.DimensionColumn;
+import com.yahoo.bard.webservice.data.metric.MetricColumn;
+import com.yahoo.bard.webservice.data.time.DefaultTimeGrain;
+import com.yahoo.bard.webservice.table.ConfigPhysicalTable;
+
+import org.joda.time.DateTimeZone;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+
+/**
+ * A factory that is used by default to support Simple (non-Composite) Physical Table.
+ * Classes that extend this abstract class:
+ *
+ * PermissivePhysicalTableFactory
+ * StrictPhysicalTableFactory
+ */
+abstract class SingleDataSourcePhysicalTableFactory implements Factory<ConfigPhysicalTable> {
+    private static final String ENTITY_TYPE = "single data source physical table";
+
+    /**
+     * Build the parameter for the subclass of SingleDataSourceParams to use.
+     *
+     * @param name  name of the LuthierTable as a String
+     * @param configTable  ObjectNode that points to the value of corresponding table entry in config file
+     * @param resourceFactories  should be the industrial park that needs to be paseed along to the builders
+     *
+     * @return  a param bean that contains information need to build the PhysicalTable
+     */
+    LuthierPhysicalTableParams buildParams(
+            String name,
+            ObjectNode configTable,
+            LuthierIndustrialPark resourceFactories
+    ) {
+        validateFields(name, configTable);
+
+        LuthierPhysicalTableParams params = new LuthierPhysicalTableParams();
+        params.tableName = new LuthierTableName(name);
+        params.timeGrain = DefaultTimeGrain
+                .valueOf(configTable.get("granularity").textValue().toUpperCase(Locale.US))
+                .buildZonedTimeGrain(DateTimeZone.forID(configTable.get("dateTimeZone").textValue()));
+        params.columns = new LinkedHashSet<>();
+
+        JsonNode dimensionsNode = configTable.get("dimensions");
+        dimensionsNode.forEach(
+                node -> params.columns.add(new DimensionColumn(resourceFactories.getDimension(node.textValue())))
+        );
+
+        JsonNode metricsNode = configTable.get("metrics");
+        metricsNode.forEach(
+                node -> params.columns.add(new MetricColumn(node.textValue()))
```

Comment:
We should be calling `resourceFactories.getMetric(node.textValue()) here as well, right? Something like:
```java
node -> {
String metricName = node.textValue();
resourceFactories.getMetric(metricName);
params.columns.add(new MetricColumn(metricName));

```
To make sure we load in the metrics.

Reply:
Actually if you look slightly above that, shouldn't we keep a dictionary for DimensionColumn as well? 
That can look pretty bulky. An alternative I can think of is to extend the `DimensionColumn` class and make a singleton class to ensure different tables will query references to a same Column object when the underlying dimension matches. 
Not sure if that is really a Java-style design choice.

Reply:
@archolewa

Reply:
reflected without caching the Column objects.